replication_manager5.sh

```diff
@@ -0,0 +1,503 @@
+# 
+# with every link a master-master rel, the content will look like:
+# 
+# +---------------+-------------+
+# | clusterSlave | clusterMaster |
+# +---------------+-------------+
+# | DC1           | DC2         |
+# | DC2           | DC1         |
+# | DC2           | DC3         |
+# | DC2           | DC4         |
+# | DC3           | DC2         |
+# | DC4           | DC2         |
+# +---------------+-------------+
+# 
+# The cluster table defines the remote cluster.  masterCandidates is a space 
+# seperated list of remote masters, either fqdn or IPs, replCreds is the 
+# fragment of the changes master command that provides authentication.  For 
+# example, DC1 could be listed as:
+#
+# +---------+----------------------------------------------+-------------------------------------------------+
+# | cluster | masterCandidates                             | replCreds                                       |
+# +---------+----------------------------------------------+-------------------------------------------------+
+# | DC1     | 172.29.110.132 172.29.110.133 172.29.110.134 | master_user='repl', master_password='repl_pass' |
+# +---------+----------------------------------------------+-------------------------------------------------+
+#
+# If you need to add a custom port, just add master_port=3307 in the replCreds column.
+#
+# This script must be call every minutes by cron on every node that you 
+# want to be a potential slave.  
+# * * * * * /path/to/replication_manager.sh > /tmp/replication_manager.out
+#
+# The list of remote masters is defined by the variable: MASTERS_LIST  (see below)
+#
+# The mysql credentials needs to be in the .my.cnf file of the user under which
+# the script run.  I needs SELECT, INSERT and UPDATE on percona.repliction 
+# and SUPER on *.*
+#
+# Finally, the script requires GTID replication to be enabled.  If you are using 
+# MariaDB GTID implementation, set the variable IS_MARIADB a few lines down
+# to 1 otherwise 0.  For MariaDB, you'll need 10.1.4+ and the following variables:
+# 
+# wsrep_gtid_mode = ON
+# wsrep_gtid_domain_id set to the the same value within a cluster and a distinct
+# value between the clusters
+# gtid_ignore_duplicates = ON
+# server-id, same value within a cluster, disctinct across clusters
+#
+#exit
+
+DEBUG_LOG="/tmp/replication_manager.log"
+if [ "${DEBUG_LOG}" -a -w "${DEBUG_LOG}" -a ! -L "${DEBUG_LOG}" ]; then
+  exec 9>>"$DEBUG_LOG"
+  exec 2>&9
+  date >&9
+  set -x
+fi
+
+if [ -f /tmp/replication_manager.off ]; then
+    echo "/tmp/replication_manager.off exists, exiting now"
+    exit
+fi
+
+#Global variables default values
+FAILED_REPLICATION_TIMEOUT=179  # 3 times the cron interval minus 1s
+
+# set it to the cluster size if you want to distribute the slave, 0 otherwise
+DISTRIBUTE_SLAVE=3 
+
+MYSQL="`which mysql` --connect_timeout=10 -B"
+
+# local override
+if [ -f "/usr/local/etc/replication_manager.cnf" ]; then
+    source /usr/local/etc/replication_manager.cnf
+fi
```

Comment:
if  /usr/local/etc/replication_manager.cnf will be a link to a malicious file the script will load it. So I would suggest to add additional check that it is not a link

Reply:
added a check to be sure we are NOT referring to a link